examples/timeline-gantt.py

```diff
@@ -0,0 +1,149 @@
+    """
+    timeline_df["Start"] = pd.to_datetime(timeline_df["Start"])
+    timeline_df["Duration"] = timeline_df["Duration"].astype(int)
+    timeline_df["Finish"] = timeline_df["Start"] + pd.to_timedelta(timeline_df["Duration"], unit="D")
+    timeline_df["Start"] = pd.to_datetime(timeline_df["Start"]).dt.date
+    timeline_df["Finish"] = pd.to_datetime(timeline_df["Finish"]).dt.date
+    return timeline_df
+
+
+app = Dash(
+    __name__,
+    external_stylesheets=[dbc.themes.SPACELAB],
+    suppress_callback_exceptions=True,
+    prevent_initial_callbacks=True,
+)
+
+app.layout = dbc.Container(
+    [
+        html.H1("Project Time Line", className="bg-primary text-white p-1 text-center"),
+        dbc.Button("Add task", n_clicks=0, id="add-row-btn", size="sm"),
+        dash_table.DataTable(
+            id="user-datatable",
+            sort_action="native",
+            columns=DATA_TABLE_SCHEMA,
+            data=get_default_table().to_dict("records"),
+            editable=True,
+            dropdown={  # limit num of Resource options for the user to select due to color limitation of 26
+                "Resource": {
+                    "clearable": False,
+                    "options": [
+                        {"label": i, "value": i} for i in ["A", "B", "C", "D"]
+                    ],
+                },
+            },
+            # makes space for the dropdown caret
+            css=[
+                {"selector": ".Select-value", "rule": "padding-right: 22px"},
+                {"selector": ".dropdown", "rule": "position: static"}  # makes dropdown visible
+            ],
+            page_size=10,
+            row_deletable=True,
+            style_data_conditional=DATA_TABLE_STYLE.get("style_data_conditional"),
+            style_header=DATA_TABLE_STYLE.get("style_header"),
+        ),
+        dcc.Graph(id="gantt-graph"),
+    ],
+    fluid=True,
+)
+
+
+def create_gantt_chart(updated_table_as_df) -> plotly.graph_objs.Figure:
+    gantt_fig = px.timeline(updated_table_as_df, x_start="Start", x_end="Finish", y="Task", color="Resource",
+                            title='Project Plan Gantt Chart')
+
+    gantt_fig.update_layout(
+        title_x=0.5,
+        font=dict(size=16),
+        yaxis=dict(title="Task", automargin=True, autorange="reversed", categoryorder="array",
+                   categoryarray=updated_table_as_df["Task"]),  # sorting gantt according to datatable
+        xaxis=dict(title=""))
+    gantt_fig.update_traces(width=0.7)
+
+    return gantt_fig
+
+
+@app.callback(
+    Output("user-datatable", "data"),
+    Output("gantt-graph", "figure"),
+    Input("user-datatable", "derived_virtual_data"),
+    Input("add-row-btn", "n_clicks"),
+)
+def update_table_and_figure(user_datatable: None or list, _) -> (list, dict):
+    """
+    This callback function returns the timeline chart and the updated datatable for the main app layout
+    """
```

Comment:
This comment is not necessary because the outputs in the callback are very discriptive

```
    Output("user-datatable", "data"),
    Output("gantt-graph", "figure"),
```